api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+                                if (classroom._id.toString() == class_id) {
+                                    assignment.classrooms.splice(assignment.classrooms.indexOf(class_id), 1, classroom);
+                                }
+                            });
+                        });
+                        db.collection(journalCollection, function (err, collection) {
+                            if (err) {
+                                return res.status(500).send({
+                                    'error': "An error has occurred",
+                                    'code': 10
+                                });
+                            }
+                            collection.find(
+                                {
+                                    _id: new mongo.ObjectID(assignment.journal_id)
+                                },
+                                {
+                                    $project: {
+                                        'content.objectId': 1,
+                                        'content.metadata': 1,
+                                        'content.text': 1,
+                                    }
+                                }
+                            ).toArray(function (err, journals) {
+                                if (err) {
+                                    return res.status(500).send({
+                                        'error': "An error has occurred",
+                                        'code': 10
+                                    });
+                                }
+                                journals.find(function (journal) {
+                                    journal.content.filter(function (entry) {
+                                        if (entry.objectId === assignment.assignedWork) {
+                                            assignment.assignedWork = entry;
+                                        }
+                                    });
+                                });
+                                res.status(200).send(assignment);
+                            });
+                        });
+                    }
+                });
+            });
+        });
+    });
+};
+
+//launch Assignment
+exports.launchAssignment = function (req, res) {
+    //validate
+    if (!mongo.ObjectID.isValid(req.params.assignmentId)) {
+        return res.status(401).send({
+            'error': "Invalid assignment id",
+            'code': 35
+        });
+    }
+    //find assignment by id
+    db.collection(assignmentCollection, function (err, collection) {
+        collection.findOne({ _id: new mongo.ObjectID(req.params.assignmentId) }, function (err, assignment) {
+            if (err) {
+                return res.status(400).send({
+                    'error': 'Inexisting assignment id',
+                    'code': 23
+                });
+            }
+            if (!assignment) {
```

Comment:
Also, why is status code 401 returned if assignment does not exists? This code is used for unauthorized requests.

Reply:
I guess we have used 401 for this purpose in other APIs in Sugarizer Server, so it cannot be avoided. But please do add error message and code.